p2p/discv5/udp.go

```diff
@@ -376,6 +368,7 @@ func encodePacket(priv *ecdsa.PrivateKey, ptype byte, req interface{}) (p, hash
 	// add the hash to the front. Note: this doesn't protect the
 	// packet in any way.
 	hash = crypto.Keccak256(packet[macSize:])
+	hash[0]++ // guarantee incompatibility between v4 and temporary v5 packet formats
```

Comment:
This makes the format incompatible with v4, but also breaks compatibility with the previous v5 format. The DHT will be split. I'm not sure that's OK. We're about to break the format anyway though, so maybe it is OK after all. I held off merging this PR because of this issue.

Reply:
Port sharing breaks compatibility anyway, not because of the format but because of the changed port address (the +1/-1 offset was hardcoded). I think if we do not break compatibility with existing v5 format that would cause hell. I want to get rid of the dual port chaos and have a fresh start. Right now the server advertising of the latest release is broken anyway (there was a bug that broke registration when registering multiple topics) so we have nothing to lose. I hope if we release this and people update then we will at least return to pre-LES/2 usability which will be fine until we can release the new and reworked discovery.

Reply:
OK